pilot/pkg/model/endpointshards.go

```diff
@@ -204,6 +204,9 @@ func (e *EndpointIndex) DeleteShard(shardKey ShardKey) {
 			e.deleteServiceInner(shardKey, svc, ns, false)
 		}
 	}
+	if e.cache == nil {
```

Comment:
Are n't we instantiating `DisabledCache` when xds cache is disabled so that we do not check at every other place?

Reply:
That is set in the Discovery Server， here it is in `EndpointIndex`, we don't set it if the cache disabled

Reply:
why do not we start with DisableCache in constructor?

Reply:
i think either works